### ISSUE TITLE
Rule proposal: `consistent-date-clone`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2437]
### Description

I just learned that `Date` constructor accepts `Date` since ES6. 

> If [Type](https://262.ecma-international.org/6.0/#sec-ecmascript-data-types-and-values)(value) is Object and value has a [[DateValue]] [internal slot](https://262.ecma-international.org/6.0/#sec-object-internal-methods-and-internal-slots), then
Let tv be thisTimeValue(value).

https://262.ecma-international.org/6.0/#sec-date-constructor

### Fail

```js
new Date(date.getTime());
```

```js
new Date(
	date.getFullYear(),
	date.getMonth(),
	date.getDate(),
	date.getHours(),
	date.getMinutes(),
	date.getSeconds(),
	date.getMilliseconds(),
);
```

### Pass

```js
new Date(date)
```

### Proposed rule name

consistent-date-clone

### Additional Info

Note: `new Date(date)` doesn't work before ES6

> Let v be [ToPrimitive](https://262.ecma-international.org/5.1/#sec-9.1)(value).
If [Type](https://262.ecma-international.org/5.1/#sec-8)(v) is String, then
Parse v as a date, in exactly the same manner as for the parse method ([15.9.4.2](https://262.ecma-international.org/5.1/#sec-15.9.4.2)); let V be the [time value](https://262.ecma-international.org/5.1/#sec-15.9.1.1) for this date.
Else, let V be [ToNumber](https://262.ecma-international.org/5.1/#sec-9.3)(v).
Set the [[PrimitiveValue]] internal property of the newly constructed object to [TimeClip](https://262.ecma-international.org/5.1/#sec-15.9.1.14)(V) and return

https://262.ecma-international.org/5.1/#sec-15.9.3.2

**Comment by sindresorhus** [comment]
Accepted

**Comment by fisker** [comment]
Since `milliseconds` exists, If we want check

```js
new Date(
	date.getFullYear(),
	date.getMonth(),
	date.getDate(),
	date.getHours(),
	date.getMinutes(),
	date.getSeconds(),
	date.getMilliseconds(),
);
```

We will need an option to ignore `milliseconds` to check

```js
new Date(
	date.getFullYear(),
	date.getMonth(),
	date.getDate(),
	date.getHours(),
	date.getMinutes(),
	date.getSeconds(),
);
```

Maybe we should ignore these two cases, only report

```js
new Date(date.getTime());
```

**Comment by fisker** [comment]
Fixed by  #2544
